api/policies/checkDownloadToken.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Nanocloud turns any traditional software into a cloud solution, without
+ * changing or redeveloping existing source code.
+ *
+ * Copyright (C) 2016 Nanocloud Software
+ *
+ * This file is part of Nanocloud.
+ *
+ * Nanocloud is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Nanocloud is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General
+ * Public License
+ * along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/* globals AccessToken,StorageService */
+
+module.exports = function(req, res, next) {
+  let filename = req.query["filename"];
+  let downloadToken = req.query["token"];
+
+  AccessToken.findOne({
+    id: downloadToken.split(":")[0]
+  }, (err, accessToken) => {
+    if (err !== null) {
+      res.negotiate(err);
```

Comment:
`res.negotiate` should be the last function ever call. Please precede every call with a `return` statement.